content/en/blog/posts/2021-08-02-adding-gatekeeper.md/index.md

```diff
@@ -0,0 +1,179 @@
+---
+date: 2021-08-02
+title: "Gatekeeper Policies with Shipwright"
+linkTitle: "Gatekeeper Policies with Shipwright"
+description: "Restricting Shipwright to trusted sources using Gatekeeper"
+author: "Blair Drummond ([@blairdrummond](https://github.com/blairdrummond))"
+resources:
+- src: "**.{png}"
+  title: "Figure #:counter"
+---
+
+
+[Gatekeeper](https://github.com/open-policy-agent/gatekeeper) is a customizable [admission webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) for Kubernetes, which allows you to configure [policy](https://www.openpolicyagent.org/docs/latest/policy-language/) over what resources can be created in the cluster. In particular, we can use Gatekeeper to add policy to Shipwright [`Build`](/docs/api/build/)s. In this example, you can see how you can use a policy to control what source repositories Shipwright is allowed to build, so that you can have more control over what code executes inside your cluster.
+
+[![Gatekeeper](opa.png)](https://github.com/securekubernetes/securekubernetes/blob/master/docs/img/opa.png)
+
+## Only allow builds of allowed sources
+
+Unless you are using a tool like [source-to-image](https://github.com/openshift/source-to-image) to build images without root, there are risks to building arbitrary Docker images within your cluster. Because of these risks, an organization may want to limit builds to trusted sources, such as specific github organizations or an internally hosted git server. This is an example of how that can be configured using Gatekeeper.
```

Comment:
In the interest of vendor neutrality, please replace `Docker images` with `Open Container Initiative container images`, like one sees with https://github.com/containers/buildah#buildah---a-tool-that-facilitates-building-open-container-initiative-oci-container-images

Reply:
Also, source to image I do not believe is the only build tool with the ability to produce images with non-root.  Or at minimum, it is an evolving space.  How about we just say "Unless you are able to build images without root privileges, ..."